python/pyspark/ml/tests/test_feature.py

```diff
@@ -842,21 +845,43 @@ def test_bucketizer(self):
 
     def test_idf(self):
         dataset = self.spark.createDataFrame(
-            [(DenseVector([1.0, 2.0]),), (DenseVector([0.0, 1.0]),), (DenseVector([3.0, 0.2]),)],
+            [
+                (DenseVector([1.0, 2.0]),),
+                (DenseVector([0.0, 1.0]),),
+                (DenseVector([3.0, 0.2]),),
+            ],
             ["tf"],
         )
-        idf0 = IDF(inputCol="tf")
-        self.assertListEqual(idf0.params, [idf0.inputCol, idf0.minDocFreq, idf0.outputCol])
-        idf0m = idf0.fit(dataset, {idf0.outputCol: "idf"})
-        self.assertEqual(
-            idf0m.uid, idf0.uid, "Model should inherit the UID from its parent estimator."
+        idf = IDF(inputCol="tf")
+        self.assertListEqual(idf.params, [idf.inputCol, idf.minDocFreq, idf.outputCol])
+
+        model = idf.fit(dataset, {idf.outputCol: "idf"})
+        # self.assertEqual(
+        #     model.uid, idf.uid, "Model should inherit the UID from its parent estimator."
```

Comment:
@wbo4958 @grundprinzip 
should we revisit https://github.com/apache/spark/pull/49569 to pass the UID?

Reply:
created [SPARK-50988](https://issues.apache.org/jira/browse/SPARK-50988) to track this issue